src/Agent.Plugins/PipelineCache/FingerprintCreator.cs

```diff
@@ -231,7 +231,7 @@ internal static void CheckKeySegment(string keySegment)
 
                     if (!fileHashes.Any())
                     {
-                        throw new FileNotFoundException("No files found.");
+                        throw new FileNotFoundException($"No files found for pattern `{keySegment}`. Enclose in double-quotes to force interpretation as a string literal.");
```

Comment:
In my scenario (that triggered this PR/issue), my key part wasn't a "pattern", but just a string that could have been a real filename. I think this will be the most common problem and so we should make sure this is reflected in the message. How about:

```
No file(s) found for key part '{keySegment}'. Enclose in double-quotes to force interpretation as a string literal.
```

Besides the wording change, I also wrapped the value in single quotes vs. back-ticks. Back ticks seem odd in general ... I think most users will expect single or double quotes as the wrappers for values in a string message.